api/src/pcapi/core/users/email/update.py

```diff
@@ -215,19 +220,37 @@ def validate_email_update_request(
     Therefore this function can be called multiple times with the same
     inputs safely.
     """
-    payload = ChangeEmailTokenContent.from_token(token)
-    current_email = payload.current_email
-    new_email = payload.new_email
-    user = users_repository.find_user_by_email(current_email)
+    try:
+        jwt_payload = utils.decode_jwt_token(encoded_token)
+    except Exception:
+        raise exceptions.InvalidToken()
+    new_token_type = "token_type" in jwt_payload
+    if new_token_type:
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Chaud pour ajouter des commentaires `# TODO` ici, pour aller plus vite à la suppression du code quand la migration est faite

Reply:
c'est fait.
et j'en profite pour introduire la PR où je fait ces changements
https://github.com/pass-culture/pass-culture-main/pull/7497